base/mysqlxx/Query.cpp

```diff
@@ -57,8 +60,24 @@ void Query::reset()
 void Query::executeImpl()
 {
     std::string query_string = query_buf.str();
-    if (mysql_real_query(conn->getDriver(), query_string.data(), query_string.size()))
-        throw BadQuery(errorMessage(conn->getDriver()), mysql_errno(conn->getDriver()));
+
+    MYSQL* mysql_driver = conn->getDriver();
+
+    auto & logger = Poco::Util::Application::instance().logger();
+    logger.trace("Query MySQL server using connection id %lu", mysql_thread_id(mysql_driver));
+    if (mysql_real_query(mysql_driver, query_string.data(), query_string.size()))
+    {
+        const auto errno = mysql_errno(mysql_driver);
+        switch (errno)
+        {
+        case 2006 /* CR_SERVER_GONE_ERROR */:
```

Comment:
Why can't we use constant names?

Reply:
I'll think on how to import client library error codes from contrib